src_c/display.c

```diff
@@ -1303,6 +1303,7 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
 
         /* ensure window is initially black */
         if (init_flip)
+            SDL_FillRect(surf, NULL, SDL_MapRGB(surf->format, 0, 0, 0));
             pg_flip_internal(state);
```

Comment:
The `if` condition needs flower brackets `{}`😄 without it only the first line is a part of the condition and the second line would execute regardless of the value of `init_flip`

Reply:
Damn, I knew a one line fix was too good to be true.